fmt/printf.h

```diff
@@ -125,7 +125,14 @@ class ArgConverter : public ArgVisitor<ArgConverter<T>, void> {
     using internal::Arg;
     typedef typename internal::Conditional<
         is_same<T, void>::value, U, T>::type TargetType;
+#if defined(_MSC_VER)
+#pragma warning(push)
+#pragma warning(disable: 4127)
+#endif
     if (sizeof(TargetType) <= sizeof(int)) {
```

Comment:
There is a function `const_check` specifically to workaround this warning. Does it work if you replace the explicit warning suppression with

```c++
if (const_check(sizeof(TargetType) <= sizeof(int))) {
```
?

Reply:
Yes, it does! Better that #ifdef guarding!

May I submit another pull request, or you could fix it by yourself?

Reply:
> May I submit another pull request, or you could fix it by yourself?

Please do. Thanks!